File: check_brackets.py
import sys

def check_balance(filename):
    with open(filename, 'r', encoding='utf-8') as f:
        content = f.read()
    
    stack = []
    line = 1
    col = 1
    for char in content:
        if char == '\n':
            line += 1
            col = 1
        else:
            col += 1
            
        if char == '{':
            stack.append(('{', line, col))
        elif char == '}':
            if not stack:
                print(f"Boton extra '}}' en linea {line}, col {col}")
                return False
            stack.pop()
            
    if stack:
        for char, l, c in stack:
            print(f"Boton '{char}' no cerrado abierto en linea {l}, col {c}")
        return False
    
    print("Balance de llaves OK")
    return True

if __name__ == "__main__":
    check_balance(sys.argv[1])

File: check_brackets.js
const fs = require('fs');
const content = fs.readFileSync('src/App.jsx', 'utf8');
let balance = 0;
let stack = [];
let lines = content.split('\n');

for (let r = 0; r < lines.length; r++) {
    let line = lines[r];
    for (let c = 0; c < line.length; c++) {
        if (line[c] === '{') {
            balance++;
            stack.push({ char: '{', line: r + 1, col: c + 1 });
        } else if (line[c] === '}') {
            balance--;
            stack.pop();
            if (balance < 0) {
                console.log(`Extra } found at line ${r + 1}, col ${c + 1}`);
                process.exit(1);
            }
        }
    }
}
if (balance > 0) {
    console.log(`Unclosed { found at line ${stack[stack.length-1].line}, col ${stack[stack.length-1].col}`);
    process.exit(1);
}
console.log("Brackets are balanced");
